devtools/onion.c

```diff
@@ -174,16 +169,17 @@ static void do_decode(int argc, char **argv, const u8 assocdata[ASSOC_DATA_SIZE]
 	tal_free(ctx);
 }
 
-static char *opt_set_ad(const char *arg, u8 *assocdata)
+static char *opt_set_ad(const char *arg, u8 **assocdata)
 {
-	if (!hex_decode(arg, strlen(arg), assocdata, ASSOC_DATA_SIZE))
+	*assocdata = tal_hexdata(NULL, arg, strlen(arg));
+	if (!*assocdata)
 		return "Bad hex string";
 	return NULL;
 }
 
-static void opt_show_ad(char buf[OPT_SHOW_LEN], const u8 *assocdata)
+static void opt_show_ad(char buf[OPT_SHOW_LEN], u8 *const *assocdata)
```

Comment:
ooh this is fancy. what's a `*const`?

Reply:
I think `const u8 *` is a constant pointer (so the address it points to can't be modified) while `u8 *const *` is a pointer to a const (so the pointee cannot be modified). I've never seen this before either, so maybe @rustyrussell can shed some light on this?